### PR TITLE
feat(breakpoint): add conditional breakpoint sign define

### PR DESCRIPTION
**Pull request by danielnehrig**
addressing feature #289

### CODE REVIEW

File: lua/dap/breakpoints.lua [hunk 1]
```diff
@@ -68,10 +68,17 @@ function M.toggle(opts, bufnr, lnum)
   if M.remove(bufnr, lnum) and not opts.replace then
     return
   end
+  local sign_name = 'DapBreakpoint'
+  if (non_empty(opts.condition)) then
+    sign_name = 'DapBreakpointCondition'
+  end
+  if (non_empty(opts.log_message)) then
+    sign_name = 'DapLogPoint'
+  end
```
Comment (danielnehrig) [inline lua/dap/breakpoints.lua:77]: is that ok like this
or should the be a check to ensure that not both properties of the table are non empty

Comment (mfussenegger) [reply lua/dap/breakpoints.lua:77]: The protocol specification allows both to be set - so this is fine. Or at least doesn't state explicitly that they shouldn't. So no reason to add artificial restrictions.

File: doc/dap.txt [hunk 2]
```diff
@@ -231,6 +231,7 @@ SIGNS CONFIGURATION
 nvim-dap uses four signs:
```
Comment (mfussenegger) [inline doc/dap.txt:231]:
```suggestion
nvim-dap uses five signs:
```

**Comment by mfussenegger** [comment]
I'm wondering if it would make sense to change the approach altogether and add multiple signs.  One for the breakpoint,  one for the log, one for the condition.  Depending on the sign column setting you see only one or all

**Comment by danielnehrig** [comment]
@mfussenegger do you have an idea why the ci failed
```
Run curl -OL https://raw.githubusercontent.com/norcalli/bot-ci/master/scripts/github-actions-setup.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  2474  100  2474    0     0   172k      0 --:--:-- --:--:-- --:--:--  172k

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Error: Process completed with exit code 2.
```

**Comment by theHamsta** [comment]
@danielnehrig the Neovim nightly is not available today. You can try again tomorrow. Actually nvim-dap could also be tested on 0.5.1 (just like nvim-treesitter which also has failing CIs today)

**Comment by danielnehrig** [comment]
@theHamsta thanks for the info

**Pull request merged**